### PR TITLE
ROC-3147 Update interest calculation

### PR DESCRIPTION
**Pull request by robertparkinson**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-3147

### Change description
Update the interest journey. This PR deals with if the user selects interest and applies the same rate across the entire amount. 

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [x] New pages included in a11y tests
- [x] Task list and task completeness checks updated
- [x] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "prepush": "yarn test && yarn test:routes"
   },
   "dependencies": {
-    "@hmcts/cmc-common-frontend": "^1.40.0",
+    "@hmcts/cmc-common-frontend": "file:/Users/robertparkinson/git/MOJ/cmc-common-frontend",
```
Comment (timja-kainos) [inline package.json]: i thought that if you did 
`yarn link @hmcts/cmc-common-frontend`
It didn't touch the package.json? It just removes the current one and adds a symlink

File: src/main/features/claim/views/check-and-send.njk [hunk 2]
```diff
@@ -23,7 +23,7 @@
           }
         )
     }}
-    {{ row('Phone:', draftClaim.claimant.mobilePhone.number, paths.claimantMobilePage.uri) }}
+    {{ row('Phone:', draftClaim.claimant.mobilePhone.number, paths.claimantMobilePage.uri, bold = true) }}
```
Comment (timja-kainos) [inline src/main/features/claim/views/check-and-send.njk]: Please separate these changes out into a different PR, there's over 2000 changes in this PR, any unrelated changes should be done separately

File: src/main/features/claim/views/check-and-send.njk [hunk 3]
```diff
@@ -38,36 +38,50 @@
...
-        {{ row('Interest rate:', draftClaim.interest.rate | numeral('0,0.00') + '\% \(' + draftClaim.interest.type + '\)' , paths.interestPage.uri) }}
+    {% if (draftClaim.interest.option === InterestOption.YES) %}
+      {{ row('Claim Interest', 'Yes' , paths.interestPage.uri, bold = true) }}
+      {% if (draftClaim.interestType.option === InterestTypeOption.SAME_RATE) %}
+        {{ row('How do you want to claim interest?', 'Same rate for the whole period' , paths.interestRatePage.uri, bottomBorder = false, bold = true) }}
       {% endif %}
-      {% if payAtSubmission === false %}
-        {{ row('Date interest is claimed from:', draftClaim.interestDate.date.asString() | date, paths.interestDatePage.uri) }}
+      {% if (draftClaim.interestRate.type) %}
+        {{ row('What rate of interest do you want to claim?', draftClaim.interestRate.rate | numeral('0,0.00') + '\%', bottomBorder = false, bold = true) }}
```
Comment (timja-kainos) [inline src/main/features/claim/views/check-and-send.njk]: is `numeral('0,0.00')` not our default format?
i.e. what happens if you do just: `numeral`

File: src/main/features/claim/views/interest-date.njk [hunk 4]
```diff
@@ -13,43 +13,34 @@
...
-              value = InterestDateType.SUBMISSION,
-              hint = ' ',
-              checked = form.valueFor('type') == InterestDateType.SUBMISSION,
-              bold = true
-            ) }}
-          </fieldset>
+          {% set name = 'type' %}
+
+          {{
+          radioGroup(
```
Comment (timja-kainos) [inline src/main/features/claim/views/interest-date.njk]: has anyone asked for GA analytics here? looks like a good candidate

File: src/main/features/response/views/confirmation.njk [hunk 5]
```diff
@@ -37,7 +37,7 @@
       <h2 class="heading-medium">{{ t('Mediation') }}</h2>
       <p>{{ t('You have chosen to use the free mediation service to resolve the dispute with {{ claimant }}. We’ll ask them if they want to mediate.', { claimant: claim.claimData.claimant.name  }) | safe }}</p>
       <p>{{ t('If they agree to mediation we’ll email you to arrange an appointment.') }}</p>
-      <p>{{ t('If you can reach an agreement with Jan Clark you won’t need to go to a hearing.') }}</p>
+      <p>{{ t('If you can reach an agreement with {{ claimant }} you won’t need to go to a hearing.', { claimant: claim.claimData.claimant.name  }) }}</p>
```
Comment (timja-kainos) [inline src/main/features/response/views/confirmation.njk]: good to fix but not the right PR

File: package.json (hunk 1 above)
Comment (robertparkinson) [reply package.json]: I'll have to give that a try. this has been the way I've always done it :)

File: src/main/features/response/views/confirmation.njk (hunk 5 above)
Comment (robertparkinson) [reply src/main/features/response/views/confirmation.njk]: I think I did it to show vivek the code on slack. I had assumed he was going to create a PR and wanted to give him the correct code to use

File: src/main/features/claim/views/check-and-send.njk (hunk 3 above)
Comment (robertparkinson) [reply src/main/features/claim/views/check-and-send.njk]: it adds a £ sign if I remove the passed in formatting

File: src/main/common/interestUtils.ts [hunk 6]
```diff
@@ -51,3 +56,19 @@ export async function draftClaimAmountWithInterest (claimDraft: DraftClaim): Pro
...
+  }
+
+  if (claimDraft.interestRate.type === InterestRateOption.STANDARD) {
+    return getStandardInterestRate()
+  } else {
+    return claimDraft.interestRate.rate
+  }
+}
+
+export function getStandardInterestRate (): number {
```
Comment (timja-kainos) [inline src/main/common/interestUtils.ts]: why do you need a function for this?
you can do an `export const` instead?
REQ

File: src/main/app/claims/models/interest.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { InterestBreakdown } from 'claims/models/interestBreakdown'
+
+export class Interest {
+
+  constructor (public type?: string, public rate?: number, public reason?: string, public specificDailyAmount?: number,
+               public interestBreakdown?: InterestBreakdown) {}
+
+  deserialize (input?: any): Interest {
+    if (input) {
+      this.type = input.type
```
Comment (dharmendrak) [inline src/main/app/claims/models/interest.ts]: type is also optional, however there is no check for it
